pkg/clients/iam/iampolicy.go

```diff
@@ -45,14 +47,44 @@ func IsPolicyUpToDate(in v1alpha1.IAMPolicyParameters, policy iamtypes.PolicyVer
 		return false, nil
 	}
 
-	compactIAMPolicy, err := awsclients.CompactAndEscapeJSON(unescapedPolicy)
+	// Compact
+	compactIn, err := awsclients.CompactJSON(in.Document)
 	if err != nil {
 		return false, err
 	}
-	compactSpecPolicy, err := awsclients.CompactAndEscapeJSON(in.Document)
+	compactExternal, err := awsclients.CompactJSON(unescapedPolicy)
+	if err != nil {
+		return false, err
+	}
+	// Normalize
+	normalizedIn := replaceActionArray(compactIn)
+	normalizedExternal := replaceActionArray(compactExternal)
+	// Escape
+	compactIAMPolicy, err := awsclients.CompactAndEscapeJSON(normalizedExternal)
+	if err != nil {
+		return false, err
+	}
+	compactSpecPolicy, err := awsclients.CompactAndEscapeJSON(normalizedIn)
 	if err != nil {
 		return false, err
 	}
 
 	return cmp.Equal(compactIAMPolicy, compactSpecPolicy), nil
 }
+
+// replaceActionArray converts Actions with a single item from an array to a string
+func replaceActionArray(compactJSON string) string {
+	// Ex.  Convert "Action": ["sts:AssumeRole"] -> "Action": "sts:AssumeRole"
+	// But ignore "Action": ["sts:AssumeRole", "sts:GetFederationToken" ] since there are multiple actions
+	r := regexp.MustCompile("\"Action\":\\[(\\S*?)\\]")
```

Comment:
I don't think a regex is suited here. The parser from #1772 could be used here as well.